packages/inter-protocol/src/auction/auctionBook.js

```diff
@@ -308,11 +328,55 @@ export const prepareAuctionBook = (baggage, zcf) => {
         /**
          * @param {Amount<'nat'>} assetAmount
          * @param {ZCFSeat} sourceSeat
+         * @param {Amount<'nat'>} [proceedsGoal] an amount that the depositor
+         *    would like to raise. The auction is requested to not sell more
+         *    collateral than required to raise that much. The auctioneer might
+         *    sell more if there is more than one supplier of collateral, and
+         *    they request inconsistent limits.
          */
-        addAssets(assetAmount, sourceSeat) {
+        addAssets(assetAmount, sourceSeat, proceedsGoal) {
           trace('add assets');
-          const { assetsForSale, collateralSeat } = this.state;
-          this.state.assetsForSale = AmountMath.add(assetsForSale, assetAmount);
+          const { collateralBrand, collateralSeat, totalProceedsGoal } =
+            this.state;
+
+          // When adding assets, the new ratio of totalCollectionGoal to collateral
+          // allocation will be the larger of the existing ratio and the ratio
+          // implied by the new deposit. Add the new collateral and raise
+          // totalProceedsGoal so it's proportional to the new ratio. This can
+          // result in raising more currency than one depositor wanted, but
+          // that's better than not selling as much as the other desired.
+
+          const allocation = collateralSeat.getCurrentAllocation();
+          const curCollateral =
+            'Collateral' in allocation
+              ? allocation.Collateral
+              : makeEmpty(collateralBrand);
+
+          // when neither proceedsGoal nor totalProceedsGoal is defined, we don't need an
+          // update and the call immediately below won't invoke this function.
+          const calcTargetRatio = () => {
+            if (totalProceedsGoal && !proceedsGoal) {
+              return makeRatioFromAmounts(totalProceedsGoal, curCollateral);
+            } else if (proceedsGoal && !totalProceedsGoal) {
+              return makeRatioFromAmounts(proceedsGoal, assetAmount);
+            } else if (proceedsGoal && totalProceedsGoal) {
+              const curRatio = makeRatioFromAmounts(
+                totalProceedsGoal,
+                curCollateral,
+              );
+              const newRatio = makeRatioFromAmounts(proceedsGoal, assetAmount);
+              return ratioGTE(newRatio, curRatio) ? newRatio : curRatio;
+            }
+          };
+
+          if (proceedsGoal || totalProceedsGoal) {
+            this.state.totalProceedsGoal = ceilMultiplyBy(
+              AmountMath.add(curCollateral, assetAmount),
+              // @ts-expect-error calcTargetRatio always returns a ratio here
```

Comment:
let's ensure that statically so this suppression isn't needed. I think a `throw Fail` at the end of `calcTargetRatio` would do it